File: common/src/main/java/com/hartwig/actin/molecular/datamodel/driver/Virus.java
package com.hartwig.actin.molecular.datamodel.driver;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class Virus implements Driver {

    @NotNull
    public abstract String name();

    public abstract boolean isReliable();

    @Nullable
    public abstract String interpretation();

    public abstract int integrations();
}

File: common/src/main/java/com/hartwig/actin/molecular/datamodel/driver/Virus.kt
package com.hartwig.actin.molecular.datamodel.driver

import com.hartwig.actin.molecular.datamodel.evidence.ActionableEvidence

data class Virus(
    val name: String,
    val type: VirusType,
    val isReliable: Boolean,
    val integrations: Int,
    override val isReportable: Boolean,
    override val event: String,
    override val driverLikelihood: DriverLikelihood?,
    override val evidence: ActionableEvidence
) : Driver
